js/ui/soundable.js

```diff
@@ -0,0 +1,68 @@
+const Lang = imports.lang;
+const St = imports.gi.St;
+const GObject = imports.gi.GObject;
+const SoundServer = imports.misc.soundServer;
+
+
+var Button = new Lang.Class({
+    Name: 'SoundableButton',
+    Extends: St.Button,
+    Properties: {
+        'click-sound-event-id': GObject.ParamSpec.string(
+            'click-sound-event-id',
+            '',
+            '',
+            GObject.ParamFlags.READWRITE,
+            ''
+        ),
+        'hover-sound-event-id': GObject.ParamSpec.string(
+            'hover-sound-event-id',
+            '',
+            '',
+            GObject.ParamFlags.READWRITE,
+            ''
+        ),
+        'stop-hover-sound-on-click': GObject.ParamSpec.boolean(
+            'stop-hover-sound-on-click',
+            '',
+            '',
+            GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
+            false
+        )
+    },
+
+    _init: function(params) {
+        this.parent(params);
+    },
```

Comment:
This method does nothing more than the default implementation now, and can be removed.